src/clients/superAgentClientImpl.ts

```diff
@@ -0,0 +1,273 @@
+import Superagent from "superagent";
+import Querystring from "querystring";
+import debug from "debug";
+import SuperAgentClient from './SuperAgentClient';
+import Logger from '../models/logger';
+
+export default class SuperAgentClientImpl implements SuperAgentClient {
+  public username: string;
+  public password: string;
+  public apiKey: string;
+  public accessToken: string;
+  public authTypes: string[];
+  public basePath: string;
+  public headerParams: object;
+  public timeout: number;
+  public pathParams: object;
+  public queryParams: object;
+  public contentType: string;
+  public formParams: object;
+  public responseType: string;
+
+  private logger: Logger;
+  private request: Superagent.SuperAgentRequest;
+
+  constructor(config: any) {
+    this.request = Superagent('', '');
+    this.username = config.username || '';
+    this.password = config.password || '';
+    this.apiKey = config.password || '';
+    this.accessToken = config.accessToken || '';
+    this.authTypes = config.authTypes || [];
+    this.basePath = config.basePath || 'https://rusticisoftware.com/api/v2/';
+    this.pathParams = config.pathParams || {};
+    this.queryParams = config.pathParams || {};
+    this.headerParams = {};
+    this.timeout = config.timeout || 60000;
+    this.contentType = config.contentType || 'application/json';
+    this.formParams = config.formParams || {};
+    this.responseType = config.responseType;
+
+    const infoLogger = debug('rustici-sdk:info');
+    const errorLogger = debug('rustici-sdk:error');
+    this.logger = {
+      info: infoLogger,
+      error: errorLogger,
+    };
+  }
+
+  public postRequest(path: string, body: object) {
+    this.request = this.createRequest('post', path, body);
+    return this.request.then(response => response);
+  }
+
+  public getRequest(path: string) {
+    this.request = this.createRequest('get', path, undefined);
+    return this.request.then(response => response);
+  }
+
+  public putRequest(path: string, body: object) {
+    this.request = this.createRequest('get', path, body);
+    return this.request.then(response => response);
+  }
+
+  public deleteRequest(path: string) {
+    this.request = this.createRequest('get', path, undefined);
+    return this.request.then(response => response);
+  }
+
+  /**
+   * Creates the SuperAgent request.
+   * 
+   * @param {string} path
+   * @param {object} queryParams 
+   * @param {object} pathParams 
+   * @param {object} body 
+   */
+  private createRequest(httpMethod: string, path: string, body: any) {
+    try {
+      const url = this.buildUrl(path, this.pathParams, '');
+      this.logger.info(url);
+
+      const request = Superagent(httpMethod, url);
+      this.applyAuthToRequest(this.authTypes);
```

Comment:
Is this right? This is applying the auth to `this.request` which was created in the constructor, not to the request created above (line 82);